ddl/reorg.go

```diff
@@ -608,7 +602,9 @@ func getReorgInfo(ctx *JobContext, d *ddlCtx, rh *reorgHandler, job *model.Job,
 		})
 
 		info.first = true
-		// get the current version for reorganization if we don't have
+		if d.lease > 0 { // Only delay when it's not in test.
+			delayForAsyncCommit()
+		}
 		ver, err := getValidCurrentVersion(d.store)
```

Comment:
There are the same codes. I think it can be merged into a single function.
```
info.first = true
...
ver, err := getValidCurrentVersion(d.store)
```

Reply:
Frankly speaking, `getReorgInfoFromPartitions` should reuse `getReorgInfo` since they are quite similar. Please note that `getReorgInfoFromPartitions` is only used by the global index feature, which is experimental, unstable, and lacks maintenance.

We need to discuss how to handle this piece of code properly.

Reply:
OK, it may be improved after GA the global index feature.